glossary.md

```diff
@@ -0,0 +1,64 @@
+# Glossary
+
+This document defines some terms that are used across this specification.
+
+<details>
+<summary>Table of Contents</summary>
+
+<!-- toc -->
+
+- [User Roles](#user-roles)
+  * [Application Author](#application-author)
+  * [Provider Author](#provider-author)
+  * [Integration Author](#integration-author)
+  * [Library Author](#library-author)
+- [Common](#common)
+  * [Feature Flag API](#feature-flag-api)
+  * [Feature Flag SDK](#feature-flag-sdk)
+  * [Provider](#provider)
+  * [Integration](#integration)
+  * [Context](#context)
+
+<!-- tocstop -->
+
+</details>
+
+## User Roles
+
+### Application Author
+
+The maintainer of an an application or service which utilizes the feature flags API to manage functionality.
+
+### Provider Author
+
+The maintainer of an SDK-compliant provider which implements the necessary interfaces required for flag evaluation.
+
+### Integration Author
+
+The maintainer of an SDK-compliant integration which implements additional secondary functionality besides flag evaluation.
+
+### Library Author
+
+The maintainer of a shared library which is a dependency of many applications or libraries, which utilizes the feature flags API to allow consumers to manage library functionality.
+
+## Common
+
+### Feature Flag API
+
+The interfaces and abstractions used by Application Author to implement feature flags in their application or service.
+
+### Feature Flag SDK
```

Comment:
Feature Flag SDK or OpenFeature SDK?

Reply:
SDK and API feel reversed here to me.

API is the thing interface that we write against. We implement the interface.
SDK is the thing we use to develop the software. The "we" in this case feels more like an end user (e.g. applicatino author) to me.

Reply:
> SDK and API feel reversed here to me.

> API is the thing interface that we write against. We implement the interface.
SDK is the thing we use to develop the software. The "we" in this case feels more like an end user (e.g. applicatino author) to me.

Thanks for the input... This is the sort of confusion I was hoping to address. I see what you're saying. I guess it's a matter of perspective.

In my mental model, the OpenFeature API is the consistent abstraction (facade) that first party application developers need to understand to use feature flags in their software. It's intellectually lightweight. The implementation (providers) are hidden from them.

The SDK is what maintainers who would work on providers and integrations would work with - it of course also has a contract that must be fulfilled, and it's a more complex one. The larger set of more complex interfaces is why I though this is really the "SDK". I do however see your point, and I dont exactly feel strongly which direction we go in, as long as it's well defined. Im open the reversing the terminology as you seem to be suggesting, or some other option.

Reply:
Here is SDK vs API from a few sources that feel fairly trustworthy.

IBM & Hubspot says they're both SDKs, and APIs are the abstract thing like a specification.
Stack overflow says the user one is an SDK and the provider one is an API.

# [IBM](https://www.ibm.com/cloud/blog/sdk-vs-api)


> SDK stands for software development kit. Also known as a devkit, the SDK is a set of software-building tools for a specific platform, including the building blocks, debuggers and, often, a framework or group of code libraries such as a set of routines specific to an operating system (OS).
> 
> API stands for application programming interface. Whether working as a standalone solution or included within an SDK, an API facilitates communication between two platforms. It does this by allowing its proprietary software to be leveraged by third-party developers. The developers can then enable their own users to indirectly use the service or services provided by the API solution.
> 
> You can also think of an API as a kind of agreement between two parties. The API not only allows for on-demand information exchange, it stipulates how that information should be exchanged.

# [Hubspot](https://blog.hubspot.com/website/sdk-vs-api)

> An SDK, or Software Development Kit, is a set of software-building tools for a specific platform, like Facebook, or programming language, like Java. This kit includes compilers, runtime environments, documentation, debuggers, and a framework or set of code libraries that are specific to the platform or language. It usually includes an API as well.
> 
> An API is an interface that enables different applications to work together. Developers typically use an API to extract information or functionality from a specific platform and use it in their own application. So, while they can’t use an API to create a completely new application, they can use it to enhance an app.


# [Stack overflow](https://softwareengineering.stackexchange.com/questions/101873/whats-the-difference-between-an-api-and-an-sdk)

> An SDK seems to be a complete set of APIs that allow you to perform most any action you would need to for creating applications. In addition an SDK may include other tools for developing for the platform/item that it is for.
> 
> An API on the other hand is just a series of related methods that may be good for a specific purpose.
> 
> As an example, the JDK (Java Development Kit) contains the API as well as the compilers, runtimes, and other miscellaneous tools. The Java API is simply all the libraries that make up the core language that you can work with out of the box.

Reply:
I would say "OpenFeature SDK/API" since that's the focus of this doc.

Reply:
I will make that change, squash the commits, and merge.

Reply:
I agree with this, but I wanted to get this merged at the moment, as @oleg-nenashev suggests, I think we can make changes incrementally.